lib/rust/ensogl/core/src/display/style/data.rs

```diff
@@ -47,16 +50,25 @@ where color::Color<C>: Into<color::Rgba>
     }
 }
 
-impl TryFrom<String> for Data {
-    type Error = ();
-    fn try_from(s: String) -> Result<Self, Self::Error> {
-        match s.parse::<f32>() {
-            Ok(t) => Ok(Data::Number(t)),
-            _ => match s.parse::<color::AnyFormat>() {
-                Ok(t) => Ok(Data::Color(t.into())),
-                _ => Err(()),
-            },
+impl From<&str> for Data {
+    fn from(t: &str) -> Data {
+        Data::Text(t.to_owned())
+    }
+}
+
+impl FromStr for Data {
+    type Err = ();
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Ok(t) = s.parse::<f32>() {
+            return Ok(Data::Number(t));
         }
+        if let Ok(t) = s.parse::<color::AnyFormat>() {
+            return Ok(Data::Color(t.into()));
+        }
+        if s.starts_with('"') && s.ends_with('"') {
+            return Ok(Data::Text(s[1..s.len() - 1].to_string()));
+        }
+        Err(())
```

Comment:
I'm a bit afraid that people would confuse converting from string with parsing the string. We could resign from converting, or make it `From<String>` (so there is a clear distinction between converting an owned string from parsing a slice). 

Regardless, there should be a clear info in `Data` how parsing works (what formats are supported, about `""` in strings etc. with examples).

Reply:
For the record: we discussed a possible alternative of removing all the `From` implementations for `Data`, and using a custom conversion trait in the `define_themes` macro instead, but decided to keep it as is just with some extra comments.

Reply:
I don't know guys what you've been talking about, but this is a serious issue for me and IMO needs to be resolved. This is confusing. Data::from("5") returns Data::String and Data::from_str("5") returns Data::Number. It should not work this way.

Reply:
I removed the `FromStr` implementation and added an explicit `Data::parse` function instead. If you have some better ideas, I'd be grateful for suggestions!